### PR TITLE
Switch to fetch() and add a test

### PR DESCRIPTION
**Pull request by carljm**
## Cute animal pic
_Because everyone likes pictures of animals._

## Description
_The commit messages say what you did; this should explain why and how._

Just sending smaller PRs for a faster review cycle. This one just converts getHarvestJSON to use `fetch()` and adds a test for it.

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally._

Set up ngrok, enabled "Oddvest - Carl Dev" on OddMinistrative board, added/removed a task from a card, everything still works.

No functional change expected from this PR.

---

## Reviewer tasks:

- [ ] Reviewed code

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ module.exports = {
   // setupFiles: [],
 
   // A list of paths to modules that run some code to configure or set up the testing framework before each test
-  // setupFilesAfterEnv: [],
+  setupFilesAfterEnv: ['./test/setup.js'],
```
Comment (carljm) [inline jest.config.js:115]: @jgerigmeyer Turns out this wasn't a good use case for `globals` setting; I was getting weird behavior where `BluebirdPromise` ended up being `undefined` if I tried to set it in `globals`. I'm not sure why but it says in the docs that `globals` has to be JSON-serializable, which I suspect might be the problem. So I just switched to a setup file.

File: test/setup.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const BluebirdPromise = require('bluebird');
```
Comment (jgerigmeyer) [inline test/setup.js]:
```suggestion
import * as BluebirdPromise from 'bluebird';
```

File: test/harvest.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { API_BASE_URL, getHarvestJSON } from '../src/lib/harvest';
+
+describe('harvest', () => {
+  test('gets JSON data from Harvest via XHR', async () => {
+    const trello = {
+      get: (scope, visibility, key) => `${scope + visibility + key}TestValue`,
+      loadSecret: (key) => `${key}TestSecret`,
+    };
+    const url = `${API_BASE_URL}some/path`;
+    fetchMock.get(url, { some: 'data' });
```
Comment (jgerigmeyer) [inline test/harvest.test.js]: I generally prefer `getOnce` to be more explicit.

```suggestion
    fetchMock.getOnce(url, { some: 'data' });
```

File: test/harvest.test.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    const data = await getHarvestJSON(trello, 'some/path');
+    expect(data).toEqual({ some: 'data' });
```
Comment (jgerigmeyer) [inline test/harvest.test.js:15]: Totally minor and entirely personal preference, but I find tests a bit easier to read if the assertions are separated by empty lines. 🤷‍♂ 

```suggestion

    expect(data).toEqual({ some: 'data' });
```

File: test/setup.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+
+window.TrelloPowerUp = { Promise: BluebirdPromise };
```
Comment (jgerigmeyer) [inline test/setup.js:4]: I think we may also want to add some cleanup for fetchMock in this file:

```
import fetchMock from 'fetch-mock';

afterEach(fetchMock.reset);
```

File: src/lib/harvest.ts [hunk 6]
```diff
@@ -1,27 +1,17 @@
...
+    ([{ accountId }, authToken]) =>
+      fetch(API_BASE_URL + path, {
+        headers: {
+          'Harvest-Account-ID': accountId,
+          Authorization: `Bearer ${authToken}`,
+          // Harvest API docs say to send this, but they don't include it in
+          // Access-Control-Allow-Headers so it's a CORS violation to send it.
+          // 'User-Agent': 'Oddvest (carl@oddbird.net)',
+        },
+      }).then((response) => response.json()),
```
Comment (jgerigmeyer) [inline src/lib/harvest.ts:16]: Not sure what this'll do if the response isn't json (e.g. network error or other), but we can handle that if we come across it.

File: src/lib/harvest.ts [hunk 7]
```diff
@@ -1,27 +1,17 @@
...
-        // Harvest API docs say to send this, but they don't include it in
-        // Access-Control-Allow-Headers so it's not possible to send it.
-        // xhr.setRequestHeader('User-Agent', 'Oddvest (carl@oddbird.net)');
-        xhr.send();
-      },
-    );
-  });
+  TrelloPromise.all([getEnableConfig(t), getAuthToken(t)]).then(
+    ([{ accountId }, authToken]) =>
+      fetch(API_BASE_URL + path, {
```
Comment (jgerigmeyer) [inline src/lib/harvest.ts:8]: It strikes me that if we're using `fetch` here we should check if Babel polyfills automatically, or we should add https://github.com/github/fetch. I can look into that.

**Review by jgerigmeyer — APPROVED** [review]
👏

File: test/setup.js (hunk 5 above)
Comment (carljm) [reply test/setup.js:4]: So this doesn't happen automatically? Not really clear what kind of separation between tests is enforced by Jest itself.

Comment (carljm) [reply test/setup.js:4]: Yup just tested and the mocks aren't cleared without this.

**Pull request merged**